beacon-chain/core/randao/randao.go

```diff
@@ -0,0 +1,23 @@
+package randao
+
+import (
+	"fmt"
+
+	"github.com/prysmaticlabs/prysm/beacon-chain/core/types"
+	v "github.com/prysmaticlabs/prysm/beacon-chain/core/validators"
+)
+
+// UpdateRandaoLayers increments the randao layer of the block proposer at the given slot.
+func UpdateRandaoLayers(state *types.BeaconState, slot uint64) (*types.BeaconState, error) {
```

Comment:
Are we missing a few steps here? Looking at the spec:

```
Let repeat_hash(x, n) = x if n == 0 else repeat_hash(hash(x), n-1).
Let proposer = state.validator_registry[get_beacon_proposer_index(state, state.slot)].
Verify that repeat_hash(block.randao_reveal, proposer.randao_layers) == proposer.randao_commitment.
Set state.randao_mix = xor(state.randao_mix, block.randao_reveal).
Set proposer.randao_commitment = block.randao_reveal.
Set proposer.randao_layers = 0.
```
I don't see where we `repeat_hash` and `xor`. If it's not ready to implement, can you please leave a comment saying that?

Reply:
ah nope, that part I think raul is covering it in his pr. This PR isnt implementing any core block operations. Otherwise the diff would be too big